### PR TITLE
Proposer RPC: helpers to get exits and slashings

### PR DESCRIPTION
**Pull request by terencechain**
This PR brings in the Capella changes of the new proposer RPC design
- Helper to retrieve slashing from pool
- Helper to retrieve exit from pool
- Helper to build empty block

### CODE REVIEW

File: beacon-chain/rpc/prysm/v1alpha1/validator/proposer_exists.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+)
+
+func (vs *Server) getExits(head state.BeaconState, slot types.Slot) []*ethpb.SignedVoluntaryExit {
+	exits := vs.ExitPool.PendingExits(head, slot, false /*noLimit*/)
+	validExits := make([]*ethpb.SignedVoluntaryExit, 0, len(exits))
+
+	for _, exit := range exits {
+		val, err := head.ValidatorAtIndexReadOnly(exit.Exit.ValidatorIndex)
+		if err != nil {
+			fmt.Println(err)
```
Comment (nisdas) [inline beacon-chain/rpc/prysm/v1alpha1/validator/proposer_exists.go:19]: why `println` ?

File: beacon-chain/rpc/prysm/v1alpha1/validator/proposer_exists.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+			log.WithError(err).Warn("Could not retrieve validator index")
+			continue
+		}
+		if err := blocks.VerifyExitAndSignature(val, head.Slot(), head.Fork(), exit, head.GenesisValidatorsRoot()); err != nil {
+			fmt.Println(err)
```
Comment (nisdas) [inline beacon-chain/rpc/prysm/v1alpha1/validator/proposer_exists.go:24]: same here

File: testing/util/block.go [hunk 3]
```diff
@@ -359,6 +359,22 @@ func generateDepositsAndEth1Data(
 	return currentDeposits[previousDepsLen:], eth1Data, nil
 }
 
+func GenerateVoluntaryExits(bState state.BeaconState, k bls.SecretKey, idx types.ValidatorIndex) (*ethpb.SignedVoluntaryExit, error) {
```
Comment (rkapka) [inline testing/util/block.go:362]: Test?

File: beacon-chain/rpc/prysm/v1alpha1/validator/proposer_exists.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/rpc/prysm/v1alpha1/validator/proposer_exists.go]: debug log, removed

File: testing/util/block.go (hunk 3 above)
Comment (terencechain) [reply testing/util/block.go:362]: added

**Review by nisdas — APPROVED** [review]

**Pull request merged**